src/battery-check/battery-check.c

```diff
@@ -0,0 +1,88 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "main-func.h"
+#include "sleep-config.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "Checks battery level to see whether there's enough charge.\n\n"
+               "   -h --help            show this help\n"
+               "      --version         show package version\n",
+               program_invocation_short_name);
+}
+
+static int version(void) {
+        printf("%s 1.0.0\n", program_invocation_short_name);
+        return 0;
+}
+
+static int parse_argv(int argc, char * argv[]) {
+
+        enum {
+                ARG_VERSION = 0x100,
+        };
+
+        static const struct option options[] = {
+                { "help",    no_argument, NULL, 'h'         },
+                { "version", no_argument, NULL, ARG_VERSION },
+                {}
+        };
+
+        int c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+
+                switch (c) {
+
+                case 'h':
+                        help();
+                        return 0;
+
+                case ARG_VERSION:
+                        return version();
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        if (optind < argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "%s takes no argumnents.",
```

Comment:
Typo for `arguments` (maybe just use the singular form?)

Reply:
> Typo for `arguments` (maybe just use the singular form?)

Wow.....
Awesome. Thank you everyone. Thank you so so much sirs.